modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/SnapshotService.scala

```diff
@@ -0,0 +1,59 @@
+package org.tessellation.infrastructure.snapshot
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.effect.Async
+import cats.effect.std.Queue
+import cats.kernel.Next
+import cats.syntax.either._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.config.SnapshotConfig
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.domain.snapshot._
+import org.tessellation.ext.fs2.switchRepeat
+import org.tessellation.schema.height.Height
+import org.tessellation.security.signature.Signed
+
+import fs2.Stream
+
+object SnapshotService {
```

Comment:
Sorry for not mentioning this earlier in my review, but I think that this should be named something like `SnapshotTrigger`. It's not really a service I think.

Reply:
@tbekas don't you think it can be a SnapshotService responsibility as well? I mean, this class can be extended for other snapshot-related things. Just thinking out loudly 🤔

Reply:
Yeah, but it may lead to creating `all-in-one` classes, for anything `Snapshot` related. It's nice to have small classes with a single responsibility.

Reply:
Actually the only public API is the pipeline :) so technically it is just a `Stream`.
Maybe I can just do `make[F[_]]: Stream[F, Either[Signed[DAGBlock], SnapshotTrigger]]` and name it differently 🤔